src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -322,6 +324,80 @@ private void ValidateLinkedAssemblies(BindingDiagnosticBag diagnostics, Cancella
             }
         }
 
+        internal void DiscoverInterceptorsIfNeeded()
+        {
+            if (!Volatile.Read(ref DeclaringCompilation.InterceptorsDiscoveryComplete))
+            {
+                discoverInterceptors();
+                Volatile.Write(ref DeclaringCompilation.InterceptorsDiscoveryComplete, true);
+            }
+
+            void discoverInterceptors()
+            {
+                var toVisit = ArrayBuilder<NamespaceOrTypeSymbol>.GetInstance();
+
+                var location = this.GlobalNamespace.GetFirstLocationOrNone();
+                if (!location.IsInSource)
+                {
+                    return;
+                }
+
+                // Search the namespaces which were indicated to contain interceptors.
+                ImmutableArray<ImmutableArray<string>> interceptorsNamespaces = ((CSharpParseOptions)location.SourceTree.Options).InterceptorsPreviewNamespaces;
```

Comment:
Are we testing when `InterceptorsPreviewNamespaces` has duplicates? Does `TryGetInterceptor()` fail in that case?

Reply:
A double traversal may occur in this case, I'll see if I can fix it.

Reply:
Basically, it feels to me like having a duplicate namespace here should be treated as if the namespace was only provided once.